src/controllers/orderController.js

```diff
@@ -0,0 +1,48 @@
+const { Order, OrderItem } = require('../models/Order');
+const { StatusCodes } = require('http-status-codes');
+const stripe = require('stripe')(process.env.STRIPE_SECRET_KEY);
+const { BadRequestError } = require('../errors');
+
+const createOrder = async (req, res) => {
+  const { orderItems, subtotal, tax } = req.body;
+
+  try {
+    // Create an array of OrderItem objects
+    const orderItemObjects = orderItems.map(item => new OrderItem(item));
+    // Calculate the total price including tax
+    const total = subtotal + (subtotal * tax);
+
+    // Perform necessary checks and validations
+    if (orderItems.length === 0) {
+      throw new BadRequestError('Unable to create order because the order items are missing.'); 
+    }
+
+    // Create an Order document in the database
+    const order = await Order.create({
+      user: req.user.userId,
+      orderItems: orderItemObjects,
+      subtotal,
+      tax,
+      total,
+      orderStatus: 'pending', // Set initial order status as pending
```

Comment:
It's fine to do this explicitly here; another option would be making the default value of this field if not provided, `pending`

Reply:
<img width="554" alt="pending" src="https://github.com/Code-the-Dream-School/dd-prac-team4-back/assets/95896159/ca903a8e-8f10-4da0-98ab-c66ce81d4126">
<img width="602" alt="controller" src="https://github.com/Code-the-Dream-School/dd-prac-team4-back/assets/95896159/78b72cad-b262-4dad-91eb-f01c1f262b6f">

Reply:
Is this changed here in this branch or in the PS089 one? I don't see the changes you screenshotted here.

Reply:
PS089, there my orderController.js. Sorry)